audit/base/base_unix.go

```diff
@@ -0,0 +1,34 @@
+//go:build unix
+
+package base
+
+import (
+	"bytes"
+	"unsafe"
+
+	"golang.org/x/sys/unix"
+)
+
+func kernelVer() (string, error) {
+	uts, err := uname()
+	if err != nil {
+		return "", err
+	}
+	trimRelease := bytes.Trim(uts.Release[:], "\x00")
+	return bytesToStr(trimRelease), nil
+}
+
+func uname() (unix.Utsname, error) {
+	uts := unix.Utsname{}
+
+	if err := unix.Uname(&uts); err != nil {
+		return uts, err
+	}
+	return uts, nil
+}
+
+// bytesToStr converts a byte slice to a string without a copy (aka no allocation).
+// This uses unsafe, so only use in cases where you know the byte slice is not going to be modified.
+func bytesToStr(b []byte) string {
```

Comment:
Duplicate code from `msgs`

You could put this into an `internal` package and share it. Right now it's a single function so maybe not worth but if there will be others as well may start being worth having some `internal/utils` package.

Reply:
Go proverb(Pike): A little copying is better than a little dependency

Because this is such a small function, I tend to put it everywhere vs a larger utils that puts pressure on the compiler.  And while I occasionally use the `utils` name, it is considered bad form in go: https://google.github.io/styleguide/go/decisions#naming

Which then has me doing something like utils/strings...  

All this to say, this is intentionally copied.